### DISCUSSION TITLE
API Web Service vs NDFD Web Services

### DISCUSSION THREAD

**Discussion by drmrbrewer** [weather-gov/api#290] [Q&A]
What is the difference between the [API Web Service](https://www.weather.gov/documentation/services-web-api) and the [NDFD Web Services](https://vlab.ncep.noaa.gov/web/mdl/ndfd-web-services)?  Both can provide a timeseries forecast for a particular lat/long via an API call, for example [this](https://graphical.weather.gov/xml/sample_products/browser_interface/ndfdXMLclient.php?lat=40.6971494&lon=-74.2598616&product=time-series&temp=temp&qpf=qpf&rh=rh&icons=icons) for the NDFD API, while for the former one there is a two-stage process, e.g. [this](https://api.weather.gov/points/40.6971494,-74.2598616) then [this](https://api.weather.gov/gridpoints/OKX/24,32/forecast/hourly).  Are they based off the same underlying data?  Which one should I be using?

**Comment by scadergit** [comment]
![image](https://user-images.githubusercontent.com/17500494/93803732-ea5c2000-fc0a-11ea-9603-9ab7fe273f76.png)

...but seriously, MDL is focused on reasearch and development, and weather.gov is focused on dissemination. Their REST service was developed to share their research, which includes a forecast product. Both are based on the forecast netcdf created by the field offices. Similarly, you can get also observations directly from MADIS, but the user experience is not optimal.

The API is more inclusive, we dissiminate all products, including text, alerts, observations, etc. using modern technologies. It was designed to be highly efficient with our CDN, which is why there are two steps. The NWS doesn't issue forecasts by points, but only by a series of grids; where hundreds of thousands of points reduce to the same forecast. Once you know what grid point you need, you no longer have to make two queries.

**Comment by drmrbrewer** [comment]
Thanks for the excellent summary, and for the LOL... I'm thankful that I'm using the right API and don't have to rewrite everything 😂
